backend/src/v1/services/external-consumer-service.ts

```diff
@@ -147,134 +72,56 @@ const externalConsumerService = {
-				        left join naics_code as naics_code on naics_code.naics_code = report.naics_code
-                left join pay_transparency_company as company on company.company_id = report.company_id
-                left join employee_count_range as employee_count_range on employee_count_range.employee_count_range_id = report.employee_count_range_id
-                where report_status = 'Published'
-                    and (report.update_date >= ${convert(startDt).toDate()}
-                         and report.update_date < ${convert(endDt).toDate()})))
-      order by update_date, revision
-      offset ${offset}
-      limit ${limit}) as reports
-
-left join
-    (select data.report_id as calculated_data_report_id,
-            data.calculation_code_id,
-            data.value,
-            data.is_suppressed,
-            code.calculation_code
-     from
-         (select data.report_id,
-                 data.calculation_code_id,
-                 data.value,
-                 data.is_suppressed
-          from pay_transparency_calculated_data as data where data.update_date >= ${convert(startDt).toDate()}
-                         and data.update_date < ${convert(endDt).toDate()}
-          union
-              (select data.report_history_id as report_id,
-                      data.calculation_code_id,
-                      data.value,
-                      data.is_suppressed
-               from calculated_data_history as data where data.update_date >= ${convert(startDt).toDate()}
-                         and data.update_date < ${convert(endDt).toDate()})) as data
-     left join calculation_code as code on code.calculation_code_id = data.calculation_code_id) as calculated_data on calculated_data.calculated_data_report_id = reports.report_change_id`;
 
-    const results = await prismaReadOnlyReplica
+    const records = await prismaReadOnlyReplica
       .$replica()
-      .$queryRaw<RawQueryResult[]>(getReportsQuery);
-    const uniqueReports: Record<string, RawQueryResult[]> = groupBy(
-      results,
-      (x) => x.report_change_id,
-    );
+      .reports_view.findMany({
+        where: whereClause,
+        include: {
+          calculated_data: {
+            select: {
+              value: true,
+              is_suppressed: true,
+              calculation_code: true,
+            },
+          },
+        },
+        take: limit,
+        skip: offset,
+        orderBy: [{ update_date: 'asc' }, { revision: 'asc' }],
+      });
 
-    const uniqueReportIds: string[] = keys(uniqueReports);
-
-    const reports = uniqueReportIds.map((id_rev) => {
-      const data = uniqueReports[id_rev];
-      return buildReport(data);
-    });
+    const totalRecordsCount = await prismaReadOnlyReplica
+      .$replica()
+      .reports_view.count({
+        where: whereClause,
+      });
 
     return {
+      totalRecords: totalRecordsCount,
       page: offset / limit,
       pageSize: limit,
-      records: reports,
+      records,
```

Comment:
The output includes "`report_change_id`". I'm concerned because `report_change_id `is not an actual column in our database and changes when reports are updated, one day the 3rd revision will have a `report_change_id`, but after it is update, the 3rd revision will have a different `report_change_id`. (I can demo this if needed). Follow up question, do we need to include the **report_history_id**, if we don't then we shouldn't include the `report_change_id`, if we do, then we need to reconsider the output.

Reply:
I dont think that we need to include the report_history_id, I think that they use the report_id and revision

Reply:
A combination of report_id and revision can be used to determine the history of a report. I will remove the report_change_id